test: fix length of R check in key_signature_tests

dgpv opened the pull request:
The code before the fix only checked the length of R value of the last
signature in the loop, and only for equality (but the length can be
less than 32)

The fixed code checks that length of the R value is less than or equal
to 32 on each iteration of the loop

dgpv committed:
Fix length of R check in test/key_tests.cpp:key_signature_tests

The code before the fix only checked the length of R value of the last
signature in the loop, and only for equality (but the length can be
less than 32)

The fixed code checks that length of the R value is less than or equal
to 32 on each iteration of the loop

The BOOST_CHECK(sig.size() <= 70) is merged with sig[3] <= 32 check,
and BOOST_CHECKs are moved outside the loop, for efficiency

theStack reviewed (approved):
ACK 0d96e2ff6fd68324dedf9b0963c775cc01c29998

laanwj commented on src/test/key_tests.cpp:
```diff
@@ -174,18 +174,16 @@ BOOST_AUTO_TEST_CASE(key_signature_tests)
         msg_hash = Hash(msg);
         BOOST_CHECK(key.Sign(msg_hash, sig));
-        found = sig[3] == 0x20;
+        BOOST_CHECK(sig[3] <= 0x20);
```
BOOST_CHECK is fairly slow, this is likely why it is not used in the inner loop directly.

dgpv replied:
Do you think it would be better to do

```
if( sig[3] > 0x20 ) {
    found_big = true;
    break;
}
```

and then `BOOST_CHECK(!found_big)` after the loop ?

Also there is `BOOST_CHECK` for `key.Sign()` inside the loop, do you think it should be taken outside the loop, too ?

laanwj replied:
Yes, I think it's better to keep it in a similar way as it is now, assign to a boolean then assert that after the loop.

> Also there is BOOST_CHECK for key.Sign() inside the loop, do you think it should be taken outside the loop, too ?

Probably!  I mean, a loop of 256 isn't that bad, I guess, but there's no reason a tl east to make things worse.

dgpv commented:
rebased

merged checks of `sig.size() <= 70` and `sig[3] <= 32` (with added comment)

moved `BOOST_CHECK`s out of the loop

fixed a comment ("less than or equal to 70", not "less than 70")

laanwj commented:
ACK 89895773b72275a620951730aef0b52e9437bc13
